From bf59b381e8196d5eb582e9802b3dc09956498880 Mon Sep 17 00:00:00 2001
From: Keerthy <j-keerthy@ti.com>
Date: Mon, 10 Feb 2020 16:59:29 +0530
Subject: [PATCH 23/25] arm: dts: k3-am65-main: Add ICSSG0/1 nodes

Add the ICCSG0/1 and child nodes.

Signed-off-by: Keerthy <j-keerthy@ti.com>
[Jan: rebased/fixed up for mainline]
Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 arch/arm/dts/k3-am65-main.dtsi | 204 +++++++++++++++++++++++++++++++++
 1 file changed, 204 insertions(+)

diff --git a/arch/arm/dts/k3-am65-main.dtsi b/arch/arm/dts/k3-am65-main.dtsi
index 4b5460ad24..cb28c411ef 100644
--- a/arch/arm/dts/k3-am65-main.dtsi
+++ b/arch/arm/dts/k3-am65-main.dtsi
@@ -369,6 +369,210 @@
 		ti,dis-chg-det-quirk;
 	};
 
+	icssg0: icssg@b000000 {
+		compatible = "ti,am654-icssg";
+		power-domains = <&k3_pds 62 TI_SCI_PD_EXCLUSIVE>;
+		reg = <0 0xb000000 0 0x80000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dma-ranges;
+
+		icssg0_mem: memories@b000000 {
+			reg = <0xb000000 0x2000>,
+			      <0xb002000 0x2000>,
+			      <0xb010000 0x10000>;
+			reg-names = "dram0", "dram1",
+				    "shrdram2";
+		};
+
+		icssg0_cfg: cfg@b026000 {
+			compatible = "syscon";
+			reg = <0xb026000 0x200>;
+		};
+
+		icssg0_iep: iep@b02e000 {
+			compatible = "syscon";
+			reg = <0xb02e000 0x1000>;
+		};
+
+		icssg0_mii_rt: mii-rt@b032000 {
+			compatible = "syscon";
+			reg = <0xb032000 0x100>;
+		};
+
+		icssg0_mii_g_rt: mii-g-rt@b033000 {
+			compatible = "syscon";
+			reg = <0xb033000 0x1000>;
+		};
+
+		icssg0_intc: interrupt-controller@b020000 {
+			compatible = "ti,am654-icssg-intc";
+			reg = <0xb020000 0x2000>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+			interrupts = <GIC_SPI 254 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 255 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 256 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 257 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 258 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 259 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 260 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 261 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "host2", "host3", "host4",
+					  "host5", "host6", "host7",
+					  "host8", "host9";
+		};
+
+		pru0_0: pru@b034000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb034000 0x4000>,
+			      <0xb022000 0x100>,
+			      <0xb022400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru0_0-fw";
+		};
+
+		rtu0_0: rtu@b004000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb004000 0x2000>,
+			      <0xb023000 0x100>,
+			      <0xb023400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu0_0-fw";
+		};
+
+		pru0_1: pru@b038000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb038000 0x4000>,
+			      <0xb024000 0x100>,
+			      <0xb024400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru0_1-fw";
+		};
+
+		rtu0_1: rtu@b006000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb006000 0x2000>,
+			      <0xb023800 0x100>,
+			      <0xb023c00 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu0_1-fw";
+		};
+
+		icssg0_mdio: mdio@b032400 {
+			compatible = "ti,davinci_mdio";
+			reg = <0xb032400 0x100>;
+			clocks = <&k3_clks 62 3>;
+			clock-names = "fck";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			bus_freq = <1000000>;
+			status = "disabled";
+		};
+	};
+
+	icssg1: icssg@b100000 {
+		compatible = "ti,am654-icssg";
+		power-domains = <&k3_pds 63 TI_SCI_PD_EXCLUSIVE>;
+		reg = <0 0xb100000 0 0x80000>;
+		#address-cells = <1>;
+		#size-cells = <1>;
+		dma-ranges;
+
+		icssg1_mem: memories@b100000 {
+			reg = <0xb100000 0x2000>,
+			      <0xb102000 0x2000>,
+			      <0xb110000 0x10000>;
+			reg-names = "dram0", "dram1",
+				    "shrdram2";
+		};
+
+		icssg1_cfg: cfg@b126000 {
+			compatible = "syscon";
+			reg = <0xb126000 0x200>;
+		};
+
+		icssg1_iep: iep@b12e000 {
+			compatible = "syscon";
+			reg = <0xb12e000 0x1000>;
+		};
+
+		icssg1_mii_rt: mii-rt@b132000 {
+			compatible = "syscon";
+			reg = <0xb132000 0x100>;
+		};
+
+		icssg1_mii_g_rt: mii-g-rt@b133000 {
+			compatible = "syscon";
+			reg = <0xb133000 0x1000>;
+		};
+
+		icssg1_intc: interrupt-controller@b120000 {
+			compatible = "ti,am654-icssg-intc";
+			reg = <0xb120000 0x2000>;
+			interrupt-controller;
+			#interrupt-cells = <1>;
+			interrupts = <GIC_SPI 262 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 263 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 264 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 265 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 266 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 267 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 268 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 269 IRQ_TYPE_LEVEL_HIGH>;
+			interrupt-names = "host2", "host3", "host4",
+					  "host5", "host6", "host7",
+					  "host8", "host9";
+		};
+
+		pru1_0: pru@b134000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb134000 0x4000>,
+			      <0xb122000 0x100>,
+			      <0xb122400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru1_0-fw";
+		};
+
+		rtu1_0: rtu@b104000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb104000 0x2000>,
+			      <0xb123000 0x100>,
+			      <0xb123400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu1_0-fw";
+		};
+
+		pru1_1: pru@b138000 {
+			compatible = "ti,am654-pru";
+			reg = <0xb138000 0x4000>,
+			      <0xb124000 0x100>,
+			      <0xb124400 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-pru1_1-fw";
+		};
+
+		rtu1_1: rtu@b106000 {
+			compatible = "ti,am654-rtu";
+			reg = <0xb106000 0x2000>,
+			      <0xb123800 0x100>,
+			      <0xb123c00 0x100>;
+			reg-names = "iram", "control", "debug";
+			firmware-name = "am65x-rtu1_1-fw";
+		};
+
+		icssg1_mdio: mdio@b132400 {
+			compatible = "ti,davinci_mdio";
+			reg = <0xb132400 0x100>;
+			clocks = <&k3_clks 63 3>;
+			clock-names = "fck";
+			#address-cells = <1>;
+			#size-cells = <0>;
+			bus_freq = <1000000>;
+			status = "disabled";
+		};
+	};
+
 	icssg2: icssg@b200000 {
 		compatible = "ti,am654-icssg";
 		power-domains = <&k3_pds 64 TI_SCI_PD_EXCLUSIVE>;
-- 
2.17.1

